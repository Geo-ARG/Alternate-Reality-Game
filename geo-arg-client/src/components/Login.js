import React, { Component } from 'react';
import { connect} from 'react-redux'
import { StyleSheet, View, Image, Text, Dimensions, BackAndroid, TouchableHighlight, AsyncStorage } from 'react-native';
import Carousel from 'react-native-looped-carousel'
import Icon from 'react-native-vector-icons/MaterialIcons';
import { saveUserLogin, saveData } from '../actions'

const Auth0Lock = require('react-native-lock');
const { width, height } = Dimensions.get('window')
const lock = new Auth0Lock({
  clientId: 'xZAFgD4PIqldvAzGrhaNZpWHswGIrC25',
  domain: 'user-login.auth0.com',
  allowedConnections: [
    "Username-Password-Authentication", "google-oauth2", "facebook"
  ],
  rememberLastLogin: true,
  socialButtonStyle: "big",
  theme: {},
  languageDictionary: {
    "title": "ARG Login"
  },
  language: "en"
});

let styles = StyleSheet.create({
  container:{
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  signInButton: {
    backgroundColor: '#cc6600',
    margin: 10,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 10,
    paddingBottom: 10,
    paddingRight: 30,
    paddingLeft: 30
  },
  imgHome: {
    flex: 1,
    width: null,
    height: null,
  },
  play: {
    fontSize: 25, color: 'white', fontWeight: 'bold'
  },
  playView: {
    alignSelf: 'center', justifyContent: 'center', alignItems: 'center'
  },
  backgroundView: {
    backgroundColor: 'rgba(0,0,0,.5)',
    width: width,
    height: height,
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute'
  },
  image: {
    flex: 1, width:'100%', height:'100%'
  }
});

class Login extends Component {
  constructor() {
    super()
    this.state = {
      username: '',
      email: '',
      size: { width, height }
    }
    this.loginForm = this.loginForm.bind(this)
  }

  _onLayoutDidChange = (e) => {
      const layout = e.nativeEvent.layout;
      this.setState({size: {width: layout.width, height: layout.height}})
   }


  loginForm(){
    AsyncStorage.getItem('dataUser', (err, result) => {
      if (result) {
        this.props.saveUserLogin(JSON.parse(result))
        this.props.navigator.push({page: 'home'})
      }
      else {
        lock.show({
          closable: true
        }, (err, profile, token) => {
          if (err) {
            console.log(err);
            return;
          }
          this.props.saveData(profile.nickname, profile.email)
          this.props.navigator.push({page: 'home'});
        });
      }
    });
  }

  render() {
    BackAndroid.addEventListener('LockBack', function() {
      return true;
    })
    return (
      <View style={styles.container} onLayout={this._onLayoutDidChange}>
        <Carousel
           delay={5000}
           style={this.state.size}
           autoplay
        >
          <Image
            style={styles.image}
            source={require('../assets/pokemon5.jpg')}
          />
          <Image
            style={styles.image}
            source={require('../assets/pokemon6.jpg')}
          />
          <Image
            style={styles.image}
            source={require('../assets/pokemon2.jpg')}
          />
          <Image
            style={styles.image}
            source={require('../assets/pokemon3.jpg')}
          />
          <Image
            style={styles.image}
            source={require('../assets/pokemon7.jpg')}
          />
        </Carousel>
        <View style={styles.backgroundView}>
          <Image
            source={require('../assets/logo.png')}
          />
          <TouchableHighlight
            style={styles.signInButton}
            underlayColor='#cc6600'
            onPress={this.loginForm}
          >
            <View style={styles.playView}>
              <Icon size={60} color="white" name="videogame-asset" />
              <Text style={styles.play}>PLAY!</Text>
            </View>
          </TouchableHighlight>
        </View>
      </View>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  saveUserLogin: (dataUserLogin) => dispatch(saveUserLogin(dataUserLogin)),
  saveData: (username, email) => dispatch(saveData(username, email))
})

export default connect(null, mapDispatchToProps)(Login)
