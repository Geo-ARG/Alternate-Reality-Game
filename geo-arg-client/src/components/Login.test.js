import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import React from 'react'
import configureStore from 'redux-mock-store'
import Login from './Login'

const middlewares = []
const mockStore = configureStore(middlewares)

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={mockStore({})}>
      <Login />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
