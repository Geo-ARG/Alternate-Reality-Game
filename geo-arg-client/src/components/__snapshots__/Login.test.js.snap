// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <RCTScrollView
      alwaysBounceHorizontal={false}
      alwaysBounceVertical={false}
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentContainerStyle={
        Array [
          Object {
            "position": "absolute",
          },
          undefined,
          Object {
            "height": 0,
            "width": 0,
          },
        ]
      }
      contentInset={
        Object {
          "top": 0,
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      pagingEnabled={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <Image
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "width": "100%",
            }
          }
          testID={undefined}
        />
        <Image
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "width": "100%",
            }
          }
          testID={undefined}
        />
        <Image
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "width": "100%",
            }
          }
          testID={undefined}
        />
        <Image
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "width": "100%",
            }
          }
          testID={undefined}
        />
        <Image
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "width": "100%",
            }
          }
          testID={undefined}
        />
        <Image
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "width": "100%",
            }
          }
          testID={undefined}
        />
        <Image
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "width": "100%",
            }
          }
          testID={undefined}
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,.5)",
        "height": 1334,
        "justifyContent": "center",
        "position": "absolute",
        "width": 750,
      }
    }
  >
    <Image
      source={1}
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#cc6600",
            "borderRadius": 10,
            "justifyContent": "center",
            "margin": 10,
            "paddingBottom": 10,
            "paddingLeft": 30,
            "paddingRight": 30,
            "paddingTop": 10,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 60,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          îŒ¸
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "bold",
            }
          }
        >
          PLAY!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
